### PR TITLE
CDF-21409 link to docs when resource validation fails

### PR DESCRIPTION
**Pull request by ronpal**
# Description

Please describe the change you have made.

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/_cdf_tk/load/_base_loaders.py [hunk 1]
```diff
@@ -166,6 +167,7 @@ class ResourceLoader(
     identifier_key: str = "externalId"
     dependencies: frozenset[type[ResourceLoader]] = frozenset()
     _display_name: str = ""
+    _doc_url: str = "https://developer.cognite.com/api/"
```
Comment (ronpal) [inline cognite_toolkit/_cdf_tk/load/_base_loaders.py:170]: Fallback/catchall url

File: cognite_toolkit/_cdf_tk/templates/_templates.py [hunk 2]
```diff
@@ -610,6 +601,22 @@ def validate(content: str, destination: Path, source_path: Path, modules_by_vari
...
+                parsed=item,
+                filepath_src=source_path,
+                filepath_build=destination,
+            ):
+                print(
+                    f"  [bold yellow]WARNING:[/] In module {source_path.parent.parent.name!r}, the resource {destination.parent.name!r} is not semantically correct."
+                )
+                if verbose:
+                    print(
+                        f"  [bold yellow]WARNING:[/] verify file format against the API specification for {destination.parent.name!r} at {loader._doc_url}"
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/templates/_templates.py:617]:
```suggestion
                        f"  [bold yellow]WARNING:[/] verify file format against the API specification for {destination.parent.name!r} at {loader.doc_url}"
```

File: cognite_toolkit/_cdf_tk/templates/_templates.py [hunk 3]
```diff
@@ -610,6 +601,22 @@ def validate(content: str, destination: Path, source_path: Path, modules_by_vari
...
+        if isinstance(parsed, dict):
+            parsed = [parsed]
+        for item in parsed:
+            if not check_yaml_semantics(
+                parsed=item,
+                filepath_src=source_path,
+                filepath_build=destination,
+            ):
+                print(
+                    f"  [bold yellow]WARNING:[/] In module {source_path.parent.parent.name!r}, the resource {destination.parent.name!r} is not semantically correct."
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/templates/_templates.py:613]: A bit hard to know what the actual file that is wrong.
```suggestion
                    f"  [bold yellow]WARNING:[/] In module {source_path.parent.parent.name!r}, the resource {destination.parent.name!r}/{destination.name} is not semantically correct."
```

**Review by doctrino — COMMENTED** [review]
In addition, I would improve the error message in `cdf-tk deploy` as well when files fail to load:
 `cgonite_toolkit/_cdf_tk/load/_base_loaders.py:line 543`

**Review by doctrino — APPROVED** [review]
Took the liberty of fixing this

got the memory address of the property, as the loader in build is not instantiated
```
  WARNING: In module 'example_pump_data_model', the resource 'data_models'/2.Pump.container.yaml is not semantically correct.
  WARNING: verify file format against the API specification for 'data_models' at <property object at 0x0000021E03A20A40>
```

I made the property into a class method instead, we could have made it a property/classmethod but that was deprecated in Python 3.11 and removed in 3.13 [ref](https://stackoverflow.com/questions/128573/using-property-on-classmethods).

File: cognite_toolkit/_cdf_tk/load/_base_loaders.py [hunk 4]
```diff
@@ -176,7 +182,7 @@ def __init__(self, client: CogniteClient):
 
     @property
     def display_name(self) -> str:
-        return self._display_name or self.api_name
+        return self._doc_base_url
```
Comment (ronpal) [inline cognite_toolkit/_cdf_tk/load/_base_loaders.py:185]:
```suggestion
        return self._display_name or self.api_name

```

**Pull request merged**